src/allmydata/web/directory.py

```diff
@@ -45,6 +46,8 @@ class BlockingFileError(Exception):
 def make_handler_for(node, client, parentnode=None, name=None):
     if parentnode:
         assert IDirectoryNode.providedBy(parentnode)
+    if IVerifyNode.providedBy(node):
+        return VerifyNodeHandler(client, node, parentnode, name)
```

Comment:
Here's what the offending test is complaining about:

![Browser screenshot](http://i.imgur.com/ozxB1t1.png)

This page is pretty broken, but it's not broken in a way that includes the unknown URI any longer. I suppose that that is the problem. FWIW the JSON output is unchanged:

```json
[
 "unknown",
 {
  "mutable": false,
  "ro_uri": "imm.lafs://immutable_from_the_future_imm"
 }
]
```